src/Common/Core/Impl/Collections/DictionaryExtensions.cs

```diff
@@ -8,6 +8,11 @@
 
 namespace Microsoft.Common.Core {
     public static class DictionaryExtensions {
+        public static void Deconstruct<TK, TV>(this KeyValuePair<TK, TV> kvp, out TK key, out TV value) {
```

Comment:
It is a duct typing notation. With this extension method, you can use tuples instead of `KeyValuePair`:
```cs
foreach (var (key, value) in dictionary) {

}
```